src/adagrams.js

```diff
@@ -1,15 +1,161 @@
+  const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+const SCORE_CHART = {
+    A: 1, 
+    B: 3, 
+    C: 3, 
+    D: 2, 
+    E: 1, 
+    F: 4, 
+    G: 2, 
+    H: 4, 
+    I: 1, 
+    J: 8, 
+    K: 5, 
+    L: 1, 
+    M: 3, 
+    N: 1, 
+    O: 1, 
+    P: 3, 
+    Q: 10, 
+    R: 1, 
+    S: 1, 
+    T: 1, 
+    U: 1, 
+    V: 4, 
+    W: 4, 
+    X: 8, 
+    Y: 4, 
+    Z: 10
+}
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+
+  let letters = {...LETTER_POOL}
+  let letterCount = Object.keys(letters).reduce(((total, letter) => total += letters[letter]), 0);
+
+  let letterArr = [];
+
+  for (let i = 0; i < 10; i++) {
+
+    let letter = getLetter(letters, letterCount);
+
+    letterArr.push(letter);
+
+    letterCount -= 1;
+
+  }
+
+  return letterArr;
+
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+
+  let letterObject = buildLetterObject(lettersInHand);
+
+  for (let letter of input) {
+    if (letterObject[letter]) {
+      letterObject[letter]--;
+    } else {
+      return false;
+    }
+  }
+  return true;
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+
+  word = word.toUpperCase()
+  
+  let total = word.length >= 7 ? 8 : 0;
```

Comment:
7 and 8 are also int literals in your code that are magic numbers. Would prefer some descriptively named variables to make your code more self-documenting.

```suggestion
  const bonusWordLength = 7;
  const bonusPoints = 8;
  let total = word.length >= bonusWordLength ? bonusPoints : 0;
```